plugin/output/file/file.go

```diff
@@ -169,12 +244,38 @@ func (p *Plugin) out(workerData *pipeline.WorkerData, batch *pipeline.Batch) {
 
 	outBuf := data.outBuf[:0]
 
+	minTs, maxTs := int64(0), int64(0)
 	for _, event := range batch.Events {
 		outBuf, _ = event.Encode(outBuf)
 		outBuf = append(outBuf, byte('\n'))
+
+		if p.config.MetaCfg.StoreMeta {
+			ts, err := event.Root.DigStrict("timestamp")
+			if err != nil {
+				p.logger.Errorf("doesn't have timestamp field: %s", err.Error())
+				continue
+			}
+			tsInt, err := ts.AsInt64()
+			if err != nil {
+				p.logger.Errorf("fime field isn't int64: %s", err.Error())
+				continue
+			}
+			if tsInt < minTs || minTs == 0 {
+				minTs = tsInt
+			}
+			if tsInt > maxTs {
+				maxTs = tsInt
+			}
+		}
 	}
 	data.outBuf = outBuf
 
+	if p.config.MetaCfg.StoreMeta {
+		p.mu.Lock()
```

Comment:
Please add comment about need for blocking

Reply:
Good poiunt.